### PR TITLE
Add enemy randomization

### PR DESCRIPTION
**Pull request by krypciak**

### CODE REVIEW

**Comment by krypciak** [comment]
Ready for merge

File: enemy-data.json [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        "buffalo-alt": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 3.5 },
+        "frobbit": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 1 },
+        "gray-frobbit": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 1 },
+        "seahorse": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 1.5 },
+        "penguin": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 2 },
+        "goat": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 3 },
+        "snowman": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 5.5 },
+        "mine-coldbug": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 0.8 },
+        "mine-runbot": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 3 },
+        "mine-diggingbot": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 6 },
```
Comment (2767mr) [inline enemy-data.json:17]: Crashes in `cold-dng.b1.control-a`

File: enemy-data.json [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        "jungle.plant": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 2 },
+        "jungle.blob": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 1 },
+        "jungle.blob-wave": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 1 },
+        "jungle.powerplant": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 4 },
+        "jungle.chicken": { "elements": [0, 0, 0, 1], "mapElements": "waveTeleport", "endurance": 4 },
+        "jungle.octopus": { "elements": [0, 0, 1, 0], "mapElements": "magnet", "endurance": 4 },
+        "jungle.blueray": { "elements": [0, 0, 1, 0], "mapElements": "teslaCoil", "endurance": 6 },
+        "jungle.ghost": { "elements": [0, 0, 0, 1], "mapElements": "compressor", "endurance": 6 },
+        "forest.bug-samurai-heat": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 2.5 },
+        "forest.bug-samurai-shock": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 2.5 },
```
Comment (2767mr) [inline enemy-data.json:46]: Crashes in `forest.path-09` though technically not in logic

File: enemy-data.json [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        "mine-diggingbot": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 6 },
+        "mine-kamikatze": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 3 },
+        "heat.scorpion": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 2 },
+        "heat.scorpion-alt": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 2 },
+        "heat.sandshark": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 2.5 },
+        "heat.sandworm": { "elements": [1, 0, 0, 0], "mapElements": "", "endurance": 5.5 },
+        "heat.drillertoise": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 3 },
+        "heat.volturbine": { "elements": [0, 0, 0, 0], "mapElements": "", "endurance": 3 },
+        "heat.moth": { "elements": [-1, 0, 0, 0], "mapElements": "pole", "endurance": 1.5 },
+        "heat.heat-golem": { "elements": [0, 1, 0, 0], "mapElements": "waterBubblePanel", "endurance": 6 },
```
Comment (2767mr) [inline enemy-data.json:26]: Crashes in `heat-dng.f3.room-04`

**Review by 2767mr — CHANGES_REQUESTED** [review]
Found some crashes by looking for `DO_ENEMY_ACTION` in `assets/data/maps`

**Comment by krypciak** [comment]
Can you provide your seed/randomizerState.json?

**Comment by 2767mr** [comment]
Almost all seeds will crash but here is one example: `4_6073387360149325`

**Comment by krypciak** [comment]
I cannot reproduce your crashing issues, but I've run into an infinite cutscene at `heat-dng.f3.room-04`
Are you on the latest commit?

**Comment by 2767mr** [comment]
I thought I was but I was on the wrong repo. I can see the same infinite loop

**Comment by 2767mr** [comment]
Btw, you should use an editor with TS support. If you already have that do `npx tsc --init`, add `"noImplicitAny": false,` to the tsconfig and then do `npx tsc` to check for errors

**Comment by krypciak** [comment]
I use neovim and I have error checking, Are there any errors in my code?

**Comment by 2767mr** [comment]
image

**Comment by krypciak** [comment]
Should be fixed

**Review by 2767mr — APPROVED** [review]

**Pull request merged**